dashboard/test/lib/pd/payment/payment_calculator_base_test.rb

```diff
@@ -207,6 +207,36 @@ class PaymentCalculatorBaseTest < ActiveSupport::TestCase
       assert_equal 6, workshop_summary.total_teacher_attendance_days
     end
 
+    test 'teacher summaries with deleted teacher account' do
+      workshop = create :pd_ended_workshop, num_sessions: 1
+
+      teacher_unqualified = create :pd_workshop_participant,
+        workshop: workshop,
+        enrolled: true,
+        in_section: true,
+        attended: true,
+        deleted_at: DateTime.now
+
+      calculator = PaymentCalculatorBase.instance
+      calculator.stubs(:teacher_qualified?).returns true
+      calculator.expects(:teacher_qualified?).with(teacher_unqualified).returns false
```

Comment:
Since there are no other participants in the workshop (no other teachers to check `teacher_qualified?`), the stub on line 221 is not needed. `teacher_qualified?` is expected to be called exactly once, with our expected teacher, and return false.